hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -798,6 +806,11 @@ public AbfsRestOperation append(final String path, final byte[] buffer,
       if (!op.hasResult()) {
         throw e;
       }
+
```

Comment:
on the topic of ex parsing, L797 will blow up with a ClassCastException if the exception caught is anything other than a AbfsRestOperationException.

so the type of exception caught can be changed to  AbfsRestOperationException

Reply:
Inline with this